spyder/plugins/editor.py

```diff
@@ -1183,6 +1164,8 @@ def register_plugin(self):
         if not editorstack.data:
             self.__load_temp_file()
         self.main.add_dockwidget(self)
+        self.main.add_to_fileswitcher(self, editorstack.tabs, editorstack.data,
+                                      icon='TextFileIcon')

```

Comment:
This should not be the name of the icon, but the icon returned by our icon manager, i.e.

    icon=ima.icon('TextFileIcon'))

Please change this to reflect that.